File: 프로그래머스/3/12907. 거스름돈/거스름돈.java
class Solution {
    final int NUMBER = 1000000007;
    public int solution(int n, int[] money) {
        int[] dp = new int[n+1];
        dp[0] = 1;
        for (int m : money) {
            for (int i = m; i <= n; i++) {
                dp[i] += dp[i-m] % NUMBER;
            }
        }
        return dp[n];
    }
}

File: 프로그래머스/3/12907. 거스름돈/거스름돈.py
def solution(n, money):
    answer = 0
    dp = [0] * (n+1)
    dp[0] = 1
        
    for coin in money:
        for price in range(coin, n+1):
            dp[price] = (dp[price] + dp[price-coin]) % 1000000007
    return dp[n]